### ISSUE TITLE
The trainer creation time doubled

### ISSUE THREAD

**Issue by RuofanKong** [ray-project/ray#18502]
### Issue Description
I am upgrading Ray from 0.8.6 to 1.3.0, and I noticed that the trainer creation time is 2x+ more now.

### System Info
- Ray version: 1.3.0 (compared against v0.8.6)
- Python version: 3.8.5
- Tensorflow version: 2.3.0
- OS: MacOS Catalina 10.15.6

### Repro Steps
Run the following code for both Ray 1.3.0 and 0.8.6, and you can consistently get the increased time cost for trainer creation for 1.3.0.

```
import ray
import ray.rllib.agents.ppo.ppo as ppo
import time


def create_ppo_trainer() -> None:
    t1 = time.time()
    ray.init()
    config = ppo.DEFAULT_CONFIG.copy()
    config["num_gpus"] = 0
    config["num_workers"] = 3
    ppo.PPOTrainer(config=config, env="CartPole-v0")
    # Simple timer to profile. we may use some more accurate tools like CProfile.
    print("time is: ", time.time() - t1)


create_ppo_trainer()
```

**NOTE: we may use profiling tool like CProfile to get more accurate results.**

**Comment by mvindiola1** [comment]
@RuofanKong that is because starting around ray  1.2.0 rllib started passing dummy data through the trainers at initialization to set up the ViewRequirements  for keys in the SampleBatch.

It does both passes with compute_actions and the loss function.

**Comment by sven1977** [comment]
I'm closing this issue for now. Feel free to re-open should you see further delays in creating the Trainers.
There is also a new test case here that confirms that it's much faster to provide either of the above-mentioned config options:

https://github.com/ray-project/ray/blob/master/rllib/agents/tests/test_trainer.py
